uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/infobar/Infobar.kt

```diff
@@ -0,0 +1,286 @@
+     *
+     * @param tintColor [Int] left button background tint.
+     */
+    fun setLeftButtonBackgroundTint(@ColorRes tintColor: Int) {
+        if (tintColor != 0) binding.infobarLeftButton.backgroundTint(tintColor)
+    }
+
+    /**
+     * Register a callback to be invoked when [Infobar] left action button is clicked.
+     *
+     * @param listener [View.OnClickListener] callback to invoke on [Infobar] left action button click.
+     */
+    fun setOnLeftButtonClickListener(listener: OnClickListener) {
+        binding.infobarLeftButton.setOnClickListener(listener)
+    }
+
+    /**
+     * Sets the left button visibility.
+     *
+     * @param visible [Boolean] true to sets the left button visible, false otherwise
+     */
+    fun setLeftButtonVisible(visible: Boolean) {
+        binding.infobarLeftButton.visibility = if (visible) VISIBLE else GONE
+    }
+
+    /**
+     * Sets the right button image resource and optionally an tint.
+     *
+     * @param imageResource [Int] to be used as right button image resource.
+     * @param tintColor [Int] optional parameter to be used as right button tint.
+     */
+    @JvmOverloads
+    fun setRightButtonImageResource(@DrawableRes imageResource: Int, @ColorRes tintColor: Int = 0) {
+        if (imageResource != 0) binding.infobarRightButton.setImageResource(imageResource)
+        if (tintColor != 0) setRightButtonImageTint(tintColor)
+    }
+
+    /**
+     * Sets the right button image tint.
+     *
+     * @param tintColor [Int] right button tint.
+     */
+    fun setRightButtonImageTint(@ColorRes tintColor: Int) {
+        if (tintColor != 0) binding.infobarRightButton.tint(tintColor)
+    }
+
+    /**
+     * Sets the right button background resource and optionally an tint.
+     *
+     * @param backgroundResource [Int] to be used as right button background resource.
+     * @param tintColor [Int] optional parameter to be used as right button background tint.
+     */
+    @JvmOverloads
+    fun setRightButtonBackgroundResource(@DrawableRes backgroundResource: Int, @ColorRes tintColor: Int = 0) {
+        if (backgroundResource != 0) binding.infobarRightButton.setBackgroundResource(backgroundResource)
+        if (tintColor != 0) setRightButtonBackgroundTint(tintColor)
+    }
+
+    /**
+     * Sets the right button background tint.
+     *
+     * @param tintColor [Int] right button background tint.
+     */
+    fun setRightButtonBackgroundTint(@ColorRes tintColor: Int) {
+        if (tintColor != 0) binding.infobarRightButton.backgroundTint(tintColor)
+    }
+
+    /**
+     * Register a callback to be invoked when [Infobar] right action button is clicked.
+     *
+     * @param listener [View.OnClickListener] callback to invoke on [Infobar] right action button click.
+     */
+    fun setOnRightButtonClickListener(listener: OnClickListener) {
+        binding.infobarRightButton.setOnClickListener(listener)
+        binding.infobarRightButton.visibility = View.VISIBLE
```

Comment:
left button to v listeneri nema, ten je vzdy visible?

Reply:
nee, to tu nemá být, to je pozústatek z varianty, kdy se visibilita nesetovala, samostatně. dám to pryč, díky ;)